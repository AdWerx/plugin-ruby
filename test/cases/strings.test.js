runCase("strings.rb");
runCase("strings.rb", { preferSingleQuotes: false }, "double-quotes.yml");
