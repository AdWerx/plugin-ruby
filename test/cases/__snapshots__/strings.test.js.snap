// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strings.rb matches expected output 1`] = `
"# frozen_string_literal: true

# rubocop:disable Lint/UnneededCopDisableDirective
# rubocop:disable Lint/Void, Layout/IndentHeredoc, Lint/UselessAssignment
# rubocop:disable Layout/SpaceInsideStringInterpolation
# rubocop:disable Lint/LiteralInInterpolation, Layout/ClosingHeredocIndentation
# rubocop:disable Layout/ClosingParenthesisIndentation
# rubocop:disable Style/StringLiteralsInInterpolation

'a' # these are CHARs
?\\\\C-a
?\\\\M-a
?\\\\M-\\\\C-a

''

'abc'

\\"abc's\\"

'abc'

\\"abc\\\\n\\"

\\"\\\\x0\\"

\\"\\\\M-\\\\C-a\\"

\\"abc #{super} abc\\"

\\"#{abc} abc\\"

\\"{\\\\\\"abc\\\\\\": \\\\\\"foo\\\\nbar\\\\\\"}\\"

# rubocop:disable Style/Semicolon
\\"abc #{foo; bar} abc\\"
# rubocop:enable Style/Semicolon

\\"abc #{de} fghi #{jkl} mno\\"

'abc' \\\\
  'def' \\\\
  'ghi'

\\"abc #{\\"abc #{abc} abc\\"} abc\\"

{ 'a' => 1 }

{ \\"a #{a}\\" => 1 }

:\\"abc#{abc}abc\\"

\`abc\`

\`super_super_super_super_super_super_super_super_super_super_super_super_s_long\`

<<-HERE
This is a straight heredoc!
HERE

<<-HERE
This is another straight heredoc, this time with interpolation!
#{interpolation}
So interpolated right now.
HERE

abc = <<-HERE
This is a straight heredoc on an assign!
HERE

<<-PARENT
This is a straight heredoc!
#{<<-CHILD
Why do I do this
CHILD
}
PARENT

<<~HERE
  This is a squiggly heredoc!
HERE

<<~HERE
    This is another squiggly heredoc, this time with interpolation!
    #{interpolation}
    So interpolated right now.
HERE

abc = <<~HERE
  This is a squiggly heredoc on an assign!
HERE

<<~PARENT
  This is a squiggly heredoc!
  #{<<~CHILD
      Why do I do this
    CHILD
}
PARENT

<<-GRAND
#{'interpolated'}
<<~PARENT
  #{'more interpolated'}
    <<-CHILD
      #{'what is going on'}
    CHILD
  #{'more interpolated'}
PARENT
#{'interpolated'}
GRAND

'abc \\"abc\\" abc'

# rubocop:enable Lint/Void, Layout/IndentHeredoc, Lint/UselessAssignment
# rubocop:enable Layout/SpaceInsideStringInterpolation
# rubocop:enable Lint/LiteralInInterpolation, Layout/ClosingHeredocIndentation
# rubocop:enable Layout/ClosingParenthesisIndentation
# rubocop:enable Style/StringLiteralsInInterpolation
# rubocop:enable Lint/UnneededCopDisableDirective
"
`;

exports[`strings.rb matches expected output 2`] = `
"# frozen_string_literal: true

# rubocop:disable Lint/UnneededCopDisableDirective
# rubocop:disable Lint/Void, Layout/IndentHeredoc, Lint/UselessAssignment
# rubocop:disable Layout/SpaceInsideStringInterpolation
# rubocop:disable Lint/LiteralInInterpolation, Layout/ClosingHeredocIndentation
# rubocop:disable Layout/ClosingParenthesisIndentation
# rubocop:disable Style/StringLiteralsInInterpolation

\\"a\\" # these are CHARs
?\\\\C-a
?\\\\M-a
?\\\\M-\\\\C-a

\\"\\"

\\"abc\\"

\\"abc's\\"

\\"abc\\"

\\"abc\\\\n\\"

\\"\\\\x0\\"

\\"\\\\M-\\\\C-a\\"

\\"abc #{super} abc\\"

\\"#{abc} abc\\"

\\"{\\\\\\"abc\\\\\\": \\\\\\"foo\\\\nbar\\\\\\"}\\"

# rubocop:disable Style/Semicolon
\\"abc #{foo; bar} abc\\"
# rubocop:enable Style/Semicolon

\\"abc #{de} fghi #{jkl} mno\\"

\\"abc\\" \\\\
  \\"def\\" \\\\
  \\"ghi\\"

\\"abc #{\\"abc #{abc} abc\\"} abc\\"

{ \\"a\\" => 1 }

{ \\"a #{a}\\" => 1 }

:\\"abc#{abc}abc\\"

\`abc\`

\`super_super_super_super_super_super_super_super_super_super_super_super_s_long\`

<<-HERE
This is a straight heredoc!
HERE

<<-HERE
This is another straight heredoc, this time with interpolation!
#{interpolation}
So interpolated right now.
HERE

abc = <<-HERE
This is a straight heredoc on an assign!
HERE

<<-PARENT
This is a straight heredoc!
#{<<-CHILD
Why do I do this
CHILD
}
PARENT

<<~HERE
  This is a squiggly heredoc!
HERE

<<~HERE
    This is another squiggly heredoc, this time with interpolation!
    #{interpolation}
    So interpolated right now.
HERE

abc = <<~HERE
  This is a squiggly heredoc on an assign!
HERE

<<~PARENT
  This is a squiggly heredoc!
  #{<<~CHILD
      Why do I do this
    CHILD
}
PARENT

<<-GRAND
#{\\"interpolated\\"}
<<~PARENT
  #{\\"more interpolated\\"}
    <<-CHILD
      #{\\"what is going on\\"}
    CHILD
  #{\\"more interpolated\\"}
PARENT
#{\\"interpolated\\"}
GRAND

\\"abc \\\\\\"abc\\\\\\" abc\\"

# rubocop:enable Lint/Void, Layout/IndentHeredoc, Lint/UselessAssignment
# rubocop:enable Layout/SpaceInsideStringInterpolation
# rubocop:enable Lint/LiteralInInterpolation, Layout/ClosingHeredocIndentation
# rubocop:enable Layout/ClosingParenthesisIndentation
# rubocop:enable Style/StringLiteralsInInterpolation
# rubocop:enable Lint/UnneededCopDisableDirective
"
`;
